From 70fd9dde227dc978ed64d3965e31e9696f28671d Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Thu, 14 Aug 2025 23:58:15 +0100
Subject: [PATCH 7/7] docker-entrypoint: always set node keystore password

BCFIPS requires 14 character long passwords, and elasticsearch by
default uses empty password for the node keystore. In the docker
entrypoint always create keystore with a default password
changeitchangeit, this will then allow autoconfiguration to work to
generate server keystore, transport key/truststore password and certs.
---
 .../docker/src/docker/bin/docker-entrypoint.sh        | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/distribution/docker/src/docker/bin/docker-entrypoint.sh b/distribution/docker/src/docker/bin/docker-entrypoint.sh
index d7b41b81bb7..9b4a14365eb 100755
--- a/distribution/docker/src/docker/bin/docker-entrypoint.sh
+++ b/distribution/docker/src/docker/bin/docker-entrypoint.sh
@@ -32,6 +32,17 @@ fi
 # is idempotent.
 source /usr/share/elasticsearch/bin/elasticsearch-env-from-file
 
+# Chainguard FIPS
+# BCFIPS requires password on keystore. Set it to KEYSTORE_PASSWORD,
+# or changeitchangeit as it needs to be 14 characters long.
+KEYSTORE_PASSWORD="${KEYSTORE_PASSWORD:-changeitchangeit}"
+if [[ ! -f /usr/share/elasticsearch/config/elasticsearch.keystore ]]; then
+  elasticsearch-keystore create -p <<EOF
+$KEYSTORE_PASSWORD
+$KEYSTORE_PASSWORD
+EOF
+fi
+
 if [[ -f bin/elasticsearch-users ]]; then
   # Check for the ELASTIC_PASSWORD environment variable to set the
   # bootstrap password for Security.
-- 
2.48.1

